src/d3a/models/market/market_structures.py

```diff
@@ -72,18 +74,21 @@ def offer_from_JSON_string(offer_string):
     object_type = offer_dict.pop("type")
     assert object_type == "Offer"
     real_id = offer_dict.pop('real_id')
+    offer_dict.pop('energy_rate', None)
     offer = Offer(**offer_dict)
     offer.real_id = real_id
     return offer
 
 
 class Bid(namedtuple('Bid', ('id', 'price', 'energy', 'buyer', 'seller',
-                             'original_bid_price', 'buyer_origin'))):
+                             'original_bid_price', 'buyer_origin', 'energy_rate'))):
     def __new__(cls, id, price, energy, buyer, seller, original_bid_price=None,
-                buyer_origin=None):
+                buyer_origin=None, energy_rate=None):
+        if energy_rate is None:
+            energy_rate = price / energy
```

Comment:
Could be used in [line](https://github.com/gridsingularity/d3a/pull/675/files#diff-286711868906f1b84a065750ab4e674fR96)  and [line](https://github.com/gridsingularity/d3a/pull/675/files#diff-286711868906f1b84a065750ab4e674fR102)

Reply:
Also [here](https://github.com/gridsingularity/d3a/pull/675/files#diff-286711868906f1b84a065750ab4e674fR102)